cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultTaskLocalizationContext.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime;
+
+import co.cask.cdap.api.TaskLocalizationContext;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Map;
+
+/**
+ * An implementation of {@link TaskLocalizationContext} that can be initialized with some localized resources.
+ */
+public class DefaultTaskLocalizationContext implements TaskLocalizationContext {
+
+  private final Map<String, File> localizedResources;
+
+  public DefaultTaskLocalizationContext(Map<String, File> localizedResources) {
+    this.localizedResources = localizedResources;
```

Comment:
Since the map will be returned through the `getAllLocalFiles` method, it's better to do a defensive immutable map copy.